nmcli: add more server examples

lzap opened the pull request:
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [ ] The page description includes a link to documentation or a homepage (if applicable).

I know this expands the page up to maximum of 8 examples, but the
problem here is that NetworkManager is used both on clients (laptops,
PCs) mainly for WiFi connection management but also on servers to manage
ethernet, bond or VLAN connections.

It's one of the quite complex commands with lots of options, I
handpicked three new examples I struggle quite a lot: static IP
configurations for both IPv4 and IPv6. These options are inconsistent
(ip4 vs ipv4) and I always need to dig in the man page.

To hit the 8 examples limit, I propose to delete the last exapmle which
is IMHO rarely used (`ip` is preferred) and it renders the very similar
output as the very first example (the only difference is that it lists
unmanaged devices).

Cheers!

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/tldr-pages/tldr?pullRequest=3816) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/tldr-pages/tldr?pullRequest=3816) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/tldr-pages/tldr?pullRequest=3816) it.</sub>

zdroid commented:
Your examples are way more informative, I support this totally. :smile:
Also, please sign the CLA.

sbrl commented on pages/linux/nmcli.md:
```diff
@@ -22,6 +22,14 @@
+- Create a static IPv4-only connection:
 
-`nmcli device status`
+`nmcli c add con-name enp1s0 ifname enp1s0 type ethernet ip4 10.0.0.7/8 gw4 10.0.0.1 ipv4.dns 10.0.0.1 ipv6.method ignore`
```
I'm pretty sure there's a long form of the `c` here. Even though it makes the line even longer, perhaps we could use it here please?
```suggestion
`nmcli connection add con-name enp1s0 ifname enp1s0 type ethernet ip4 10.0.0.7/8 gw4 10.0.0.1 ipv4.dns 10.0.0.1 ipv6.method ignore`
```

lzap replied:
I wanted to avoid putting more options as it's long enough. Ignore simply does not configure IPv6.

sbrl replied:
Ah, I see. Considering IPv6 is the future version of the Internet Protocol, I'm reluctant to go ahead with the `ipv6.method {{ignore}}` bit here unless there's a convincing and common use-case.

This said, it looks like you've made some pretty extensive changes since I last reviewed, so I'll do a fresh review.

lzap commented:
Apologies for the delay. I've incorporated the change. On top of that:

* Removed `con-name` from the examples as it's optional flag and NM will figure out a reasonable connection name.
* Reordered options so they are in the same order.
* Broken nmcli into subcommands
* Put the `device` list command back since there is enough room now.

agnivade commented on pages/linux/nmcli-connection.md:
```diff
@@ -0,0 +1,27 @@
+# nmcli
```
nmcli connection

lzap replied:
Good catch, rebased.

agnivade commented on pages/linux/nmcli-device.md:
```diff
@@ -0,0 +1,15 @@
+# nmcli
```
nmcli device

lzap replied:
Good catch, rebased.

sbrl commented on pages/linux/nmcli-connection.md:
```diff
@@ -0,0 +1,27 @@
+
+`nmcli connection down uuid {{uuid}}`
+
+- Create a static IPv4-only connection:
```
Perhaps we could alter the order of these 3 examples to be more similar to the order people might use them in. What about reversing it, like this?

 - Create an auto-configured.....
 - Static IPv6
 - Static IPv4

lzap replied:
I am coming from enterprise datacenters world, I can tell you that IPv6 is not a thing there yet. And at homes, that's disaster. But sure, won't hurt, shaked it a bit.

sbrl replied:
I see. With IPv6 being the future, it seems like a sensible idea to emphasise that in tldr-pages.

sbrl commented on pages/linux/nmcli-device.md:
```diff
@@ -0,0 +1,15 @@
+
+> Hardware device management with NetworkManager.
+
+- Print statuses of network interfaces:
```
Perhaps this will read better?
```suggestion
- Print the statuses of all network interfaces:
```

lzap replied:
Yeah.

sbrl commented on pages/linux/nmcli-device.md:
```diff
@@ -0,0 +1,15 @@
+
+- Connect to the Wi-Fi network with a specified name and password:
+
+`nmcli device wifi connect {{name}} password {{password}}`
```
Perhaps we could tweak this slightly to be more intuitive?
```suggestion
`nmcli device wifi connect {{ssid}} password {{password}}`
```

lzap replied:
Done.

sbrl commented on pages/linux/nmcli.md:
```diff
@@ -2,26 +2,18 @@
+`nmcli {{command}} --help`
 
-- Activate a connection by specifying an uuid:
+- Execute nmcli command:
```
Perhaps this will read better?
```suggestion
- Execute an `nmcli` command:
```

lzap replied:
Yup.

lzap commented:
> thanks so much for the edits here!

Bad on my side, I have about 200+ PRs to review on my plate these days. Slowly working it through :-)

I will add new commits, we tend to rebase as github now shows changes anyway even if you rebase. But sure, if that's your thing! Pushed.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/tldr-pages/tldr?pullRequest=3816) <br/>All committers have signed the CLA.

sbrl commented:
Merging, since @ivanhercaz's comment appears to be addressed.
